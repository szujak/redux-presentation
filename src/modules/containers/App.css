html {
  font-size: 12px;
  font-family: 'Roboto', sans-serif;
  line-height: 1;
}
