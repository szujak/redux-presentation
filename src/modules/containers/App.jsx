import 'normalize.css'
import axios from 'axios'
import { connect } from 'react-redux'
import React, { Component, Fragment } from 'react'

import './App.css'
import Form from '../components/Form'
import DataPreview from '../components/DataPreview'
import {
  getData,
  getDataSuccess,
  getDataFailure
} from '../store/actions/data-actions'

class App extends Component {
  constructor (props, context) {
    super(props, context)

    this.onFormSubmit = this.onFormSubmit.bind(this)
  }

  onFormSubmit (username) {
    this.props.getData(username)

    axios
      .get(`https://api.github.com/users/${username}`)
      .then(response => {
        this.props.getDataSuccess(response.data)
      })
      .catch(error => {
        this.props.getDataFailure(error)
      })
  }

  render () {
    return (
      <Fragment>
        <Form
          onSubmit={this.onFormSubmit}
        />
        <DataPreview
          data={this.props.appData.data}
          error={this.props.appData.error}
        />
      </Fragment>
    )
  }
}

const mapStateToProps = (state) => ({
  appData: state.appData
})

const mapDispatchToProps = (dispatch) => ({
  getData: (username) => dispatch(getData(username)),
  getDataSuccess: (data) => dispatch(getDataSuccess(data)),
  getDataFailure: (error) => dispatch(getDataFailure(error))
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
