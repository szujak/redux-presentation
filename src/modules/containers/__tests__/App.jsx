import React from 'react'
import renderer from 'react-test-renderer'
import App from '../App'

it('Renders correctly', () => {
  expect(
    renderer
      .create(<App />)
      .toJSON()
  )
    .toMatchSnapshot()
})
