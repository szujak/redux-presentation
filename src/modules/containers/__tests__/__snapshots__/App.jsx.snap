// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<p
  className={undefined}
>
  a
</p>
`;
